ecosystem/ffmpeg_plugin/kahawai_common.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Kahawai common struct and functions
+ * Copyright (c) 2023 Intel
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <mtl/st_convert_api.h>
+#include <mtl/st_pipeline_api.h>
+
+#include "kahawai_common.h"
+#include "libavutil/common.h"
+
+const KahawaiFpsDecs fps_table[] = {{ST_FPS_P50, 5000 - 100, 5000 + 100},
+                                    {ST_FPS_P29_97, 2997 - 100, 2997 + 100},
+                                    {ST_FPS_P25, 2500 - 100, 2500 + 100},
+                                    {ST_FPS_P60, 6000 - 100, 6000 + 100},
+                                    {ST_FPS_P30, 3000 - 100, 3000 + 100},
+                                    {ST_FPS_P24, 2400 - 100, 2400 + 100},
+                                    {ST_FPS_P23_98, 2398 - 100, 2398 + 100},
+                                    {ST_FPS_P119_88, 11988 - 100, 11988 + 100}};
+
+static mtl_handle shared_st_handle = NULL;
+unsigned int active_session_cnt = 0;
```

Comment:
Can we find a way to set `active_session_cnt` to static also? Also seems active_session_cnt is for counter usage, it should be declared as an atomic for multi-thread safety.

Reply:
active_session_cnt is changed in write_header constructor and write_trailer destroyer functions which are atomic and single threaded so no change here